packages/react-native/src/error-tracking/index.ts

```diff
@@ -35,4 +51,70 @@ export class ErrorTracking {
       ...additionalProperties,
     } as unknown as PostHogEventProperties)
   }
+
+  private resolveAutocaptureOptions(autocapture: AutocaptureOptions | boolean = false): AutocaptureOptions {
+    if (typeof autocapture === 'boolean') {
+      return {
+        uncaughtExceptions: autocapture,
+        unhandledRejections: autocapture,
+      }
+    }
+    return {
+      uncaughtExceptions: !!autocapture.uncaughtExceptions,
+      unhandledRejections: !!autocapture.unhandledRejections,
+    }
+  }
+
+  private autocaptureUncaughtErrors() {
+    const globalHandler = ErrorUtils.getGlobalHandler()
+    ErrorUtils.setGlobalHandler((error, isFatal) => {
+      const hint: EventHint = {
+        mechanism: {
+          type: 'onuncaughtexception',
+          handled: false,
+        },
+      }
+      const additionalProperties: any = {}
+      if (isFatal) {
+        additionalProperties['$exception_level'] = 'fatal' as SeverityLevel
+      }
+      this.captureException(error, additionalProperties, hint)
+      globalHandler?.(error, isFatal)
+    })
+  }
+
+  private autocaptureUnhandledRejections() {
+    const onUnhandledRejection = (error: unknown) => {
+      const hint: EventHint = {
+        mechanism: {
+          type: 'onunhandledrejection',
+          handled: false,
+        },
+      }
+      this.captureException(error, {}, hint)
```

Comment:
a concern that i have here is that `captureException` wont have enough time to capture the event for fatal errors
if we have the SDK as flushAt=20 default, the queue has 19, then theres an uncaught exception, the exception error is the last one to be processed, by then the process died and we lost the most important item of the queue
for uncaught errors, we should also call flush and await the promise if possible.
or we should do a sort of priority queue.
its ok to start as is but this should be a follow up since its very important for mobile error tracking